watchdog/src/remote_api/bitcoin_canister.rs

```diff
@@ -0,0 +1,123 @@
+use crate::remote_api::http::{build_transform_context, create_request, fetch_body};
+use crate::remote_api::storage;
+use crate::types::BlockHeight;
+use ic_cdk::api::management_canister::http_request::{
+    CanisterHttpRequestArgument, HttpResponse, TransformArgs,
+};
+use regex::Regex;
+use std::cell::RefCell;
+
+const RE_PATTERN: &str = r"\n\s*main_chain_height (\d+) \d+\n";
+
+// This is a thread-local storage for calculating the regex only once.
+thread_local! {
+    static REGEX: RefCell<Result<Regex, regex::Error>> = RefCell::new(Regex::new(RE_PATTERN));
+}
+
+/// Apply regex rule to a given text.
+fn apply(re: &Regex, text: &str) -> Result<String, String> {
+    match re.captures(text) {
+        None => Err("Regex: no match found.".to_string()),
+        Some(cap) => match cap.len() {
+            2 => Ok(String::from(&cap[1])),
+            x => Err(format!("Regex: expected 1 group exactly, provided {}.", x)),
+        },
+    }
+}
+
+/// The transform function for the remote API.
+#[ic_cdk_macros::query]
+fn transform_bitcoin_canister(raw: TransformArgs) -> HttpResponse {
+    let mut response = HttpResponse {
+        status: raw.response.status.clone(),
+        ..Default::default()
+    };
+    if response.status == 200 {
+        let body =
+            String::from_utf8(raw.response.body).expect("Raw response is not UTF-8 encoded.");
+        response.body = BitcoinCanister::transform(body).as_bytes().to_vec();
+    } else {
+        crate::print(&format!("Received an error: err = {:?}", raw));
+    }
+    response
+}
+
+pub struct BitcoinCanister {}
+
+impl BitcoinCanister {
+    /// The host of the remote API.
+    pub fn host() -> &'static str {
+        "ghsi2-tqaaa-aaaan-aaaca-cai.raw.ic0.app"
```

Comment:
nits: I'd make the bitcoin canister ID part of the configuration of this canister, in case we want to run a similar watchdog for the bitcoin testnet canister, for example.

The `raw.ic0.app` isn't secure. Maybe we can add a note/todo to replace this with a more secure alternative in the future? We've received requests before to expose the height/tip block hash and other info as an endpoint in the bitcoin canister, so we could consider adding that in the future.

Reply:
Done.